presto-docs/src/main/sphinx/security/salesforce.rst

```diff
@@ -0,0 +1,79 @@
+
+This allows you to enable users to authenticate to Presto via their Salesforce
+basic credentials.
+
+.. note::
+    This is *not* a Salesforce connector, and does not allow users to query
+    Salesforce data. It's simply a means by which users can authenticate to Presto, similar
+    to :doc:`ldap` or :doc:`password-file`.
+
+Salesforce Authenticator Configuration
+--------------------------------------
+
+Enable Salesforce authentication by creating an
+``etc/password-authenticator.properties`` file on the coordinator:
+
+.. code-block:: properties
+
+    password-authenticator.name=salesforce
+    salesforce.allowed-organizations=allowed salesforce org id's, or the word all
+
+The following configuration properties are available:
+
+====================================   ==============================================
+Property                               Description
+====================================   ==============================================
+``salesforce.allowed-organizations``   Comma separated list of 18 character Salesforce.com
+                                       Organization IDs for a second, simple layer of security.
+                                       This option can be explicitly ignored using ``all``, which
+                                       bypasses any check of the authenticated user's
+                                       Salesforce.com Organization ID.
+
+``salesforce.cache-size``              Max number of cached authenticated users.
+                                       Defaults to ``4096``.
+
+``salesforce.cache-expire-duration``   Duration for an entry to remain in the cache after the
+                                       first write before it is removed.
+                                       Defaults to ``2m``.
+====================================   ==============================================
+
+Salesforce Concepts
+-------------------
+There are two Salesforce specific aspects to this authenticator.  They are the use of the
+Salesforce security token, and configuration of one or more Salesforce.com Organization IDs.
+
+.. _security tokens:
+
+Security Token
+^^^^^^^^^^^^^^
+
+Credentials are a user's Salesforce username and password if Presto is connecting from a whitelisted
+IP, or username and password/`security token <https://help.salesforce.com/articleView?id=user_security_token.htm&type=5>`_
+concatenation otherwise.  For example, if Presto is *not* whitelisted, and your password is ``password``
+and security token is ``token``, use ``passwordtoken`` to authenticate.
+
+You can configure a public IP for Presto as a trusted IP by `whitelisting an IP range
+<https://help.salesforce.com/articleView?id=security_networkaccess.htm&type=5>`_.
+
+Salesforce.com Organization IDs
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can configure a one or more 18 character Salesforce.com Organization IDs. They can be used as a
```

Comment:
a one or more .. something is weird here

Reply:
You can configure one or more Salesforce Organization IDs for additional authentication. The IDs need to be 18 character long.